### PR TITLE
Changes done for Leave Community Functionality

### PR DESCRIPTION
**Pull request by anusha880**
1) Fixed JOIN Community link under particular community is fixed
2) Fix for Leave Community Link display and CSS
3) Changes to Hide leave community link when user is already in the community
4) Leave Community Funcationlity
5) Community Feed Box defect when user is not joined to particular community.

### CODE REVIEW

File: src/redux/actions/dataActions.js [hunk 1]
```diff
@@ -113,6 +149,27 @@ export const getRecommendedCommunity = () => async (dispatch, getState) => {
...
 
+  // Join Community From Community Details
+  export const leaveCommunityFromDetails = (newCommunity) => async (dispatch, getState) => {
+    dispatch({ type: LOADING_JOIN_COMMUNITY, payload: true });
+    try {
+      const {
+        user: { userInfo },
+      } = getState();
+      const result = await leaveMemberFromCommunity(userInfo, newCommunity);
+      await Promise.all([
```
Comment (Binilkks) [inline src/redux/actions/dataActions.js:160]: Why do you need promise.all here ?

File: src/redux/actions/dataActions.js [hunk 2]
```diff
@@ -101,6 +115,28 @@ export const getRecommendedCommunity = () => async (dispatch, getState) => {
...
+// Join Community From Community Details
+  export const joinCommunityFromDetails = (newCommunity) => async (dispatch, getState) => {
+    dispatch({ type: LOADING_JOIN_COMMUNITY, payload: true });
+    try {
+      const {
+        user: { userInfo },
+      } = getState();
+      const result = await joinACommunity(userInfo, newCommunity);
+      await Promise.all([
+        dispatch(getPostsOfCommunity(newCommunity)),
```
Comment (Binilkks) [inline src/redux/actions/dataActions.js:139]: Promise.all required here ?

File: src/firebaseActions/dataServices.js [hunk 3]
```diff
@@ -201,6 +201,28 @@ const joinACommunity = async (user, communityId) => {
     console.log("No such document!");
   }
 };
+
+
+const leaveMemberFromCommunity = async (user, communityId) => {
+  const { email, firstName, imageUrl } = user;
+  let membersArray = [];
+  const docRef = doc(db, "community", communityId);
+  const postRef = collection(db, "posts");
```
Comment (Binilkks) [inline src/firebaseActions/dataServices.js:210]: Why post ref needed ?

File: src/components/BGCCommunity/Leftside.js [hunk 4]
```diff
@@ -1,190 +1,252 @@
...
+}) => {
+  const { email } = userInfo;
+
+  const joinCommunityHandler = () => {
+    joinCommunityFromDetails(currentCommunityId);
+  };
+
+  const communityLeaveHandler = () => {
+    leaveCommunityFromDetails(currentCommunityId);
+    console.log("Delete CommunityID", currentCommunityId);
```
Comment (Binilkks) [inline src/components/BGCCommunity/Leftside.js:36]: Remove console logs

File: src/redux/actions/dataActions.js (hunk 1 above)
Comment (anusha880) [reply src/redux/actions/dataActions.js:160]: No, we don't require them. will remove in next commit

File: src/components/BGCCommunity/Leftside.js (hunk 4 above)
Comment (anusha880) [reply src/components/BGCCommunity/Leftside.js:36]: Sorry, will remove them.

**Pull request merged**